stdlib/public/stubs/Stubs.cpp

```diff
@@ -98,7 +107,7 @@ extern "C" uint64_t swift_uint64ToString(char *Buffer, intptr_t BufferLength,
                             /*Negative=*/false);
 }
 
-#if defined(__APPLE__) || defined(__FreeBSD__)
+#if defined(__APPLE__) || defined(__FreeBSD__) || defined(__CYGWIN__)
 static inline locale_t getCLocale() {
```

Comment:
I think if you add an exclusion for Cygwin and avoid defining this function on Cygwin, you can avoid having to define `typedef void *locale_t;` above.